framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -593,195 +484,306 @@ public void standardizeVariable(AnnotatedTypeMirror type, Element elt) {
+     *       {@code stringToJavaExpr}.
+     *   <li>{@link #buildAnnotation(AnnotationMirror, Map)}: to change the annotation returned by
+     *       this method.
+     * </ul>
+     *
+     * @param stringToJavaExpr function that converts strings to {@code JavaExpression}s
+     * @param anno annotation mirror
+     * @return an annotation created by applying {@code stringToJavaExpr} to all expressions strings
+     *     in {@code anno}
      */
-    private void standardizeAtm(
-            JavaExpressionContext context,
-            TreePath localVarPath,
-            AnnotatedTypeMirror type,
-            boolean removeErroneousExpressions) {
-        this.standardizeTypeAnnotator.init(context, localVarPath, removeErroneousExpressions);
-        this.standardizeTypeAnnotator.visit(type);
+    protected @Nullable AnnotationMirror map(
+            StringToJavaExpression stringToJavaExpr, AnnotationMirror anno) {
+        if (!isExpressionAnno(anno)) {
+            return null;
+        }
+
+        Map<String, List<JavaExpression>> map = new HashMap<>();
+        for (String value : getListOfExpressionElements(anno)) {
+            List<String> expressionStrings =
+                    AnnotationUtils.getElementValueArray(anno, value, String.class, true);
+            List<JavaExpression> javaExprs = new ArrayList<>(expressionStrings.size());
+            map.put(value, javaExprs);
+            for (String expression : expressionStrings) {
+                JavaExpression result;
+                if (shouldParseExpression(expression)) {
+                    try {
+                        result = stringToJavaExpr.toJavaExpression(expression);
+                    } catch (JavaExpressionParseException e) {
+                        result = createError(expression, e);
+                    }
+                } else {
+                    result = passThroughString(expression);
+                }
+
+                if (result != null) {
+                    result = transform(result);
+                    javaExprs.add(result);
+                }
+            }
+        }
+        return buildAnnotation(anno, map);
     }
 
     /**
-     * Standardizes a Java expression.
+     * This method is for subclasses to override to change JavaExpressions in some way. This
+     * implementation returns the argument. This method is called after parsing and
+     * viewpoint-adaption have occurred. {@code javaExpr} may be a {@link PassThroughExpression}
+     * that results from an {@link JavaExpressionParseException}.
      *
-     * @param expression a Java expression
-     * @param context the context
-     * @param localVarPath if non-null, the expression is parsed as if it were written at this
-     *     location
-     * @return the standardized version of the Java expression
+     * @param javaExpr a JavaExpression
+     * @return a transformed JavaExpression or {@code null} if no transformation exists
```

Comment:
If null is returned, is the original used, or is the original discarded?  Or is that up to the caller?